src/panels/alarm-table/alarm_details.js

```diff
@@ -270,14 +270,16 @@ export class AlarmDetailsCtrl {
     let sortedFeedback = _.orderBy(feedback, ['timestamp'],  ['desc']);
     for (let fb of sortedFeedback) {
       const index = this.$scope.situationFeedback.findIndex(ifb => ifb.alarmKey === fb.alarmKey);
-      if (this.$scope.situationFeedback[index].timestamp < fb.timestamp) {
-        this.$scope.situationFeedback[index].rootCause = fb.rootCause;
-        this.$scope.situationFeedback[index].tags = fb.tags;
-        this.$scope.situationFeedback[index].feedbackType = fb.feedbackType;
-        this.$scope.situationFeedback[index].timestamp = fb.timestamp;
-        for (let tag of fb.tags) {
-          this.$scope.feedbackTags.add(tag);
-          this.$scope.tagsInput.tagsinput('add', tag);
+      if (index >= 0 && index <= this.$scope.situationFeedback.length) {
```

Comment:
We don't need to check for `index <= this.$scope.situationFeedback.length`. 

I'd replace this the following check instead of indenting the whole block:
```
if (index < 0) {
        // We have feedback for an alarm that no longer exists in the situation, ignore it.
        continue;
      }
```